engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala

```diff
@@ -117,6 +118,38 @@ case class WorkflowStoreActor(store: WorkflowStore, serviceRegistryActor: ActorR
     goto(Working) using nextData
   }
 
+  private def storeWorkflowSources(sources: NonEmptyList[WorkflowSourceFiles]): Future[NonEmptyList[WorkflowId]] = {
+    for {
+      processedSources <- Future.fromTry(processSources(sources, _.asPrettyJson))
+      workflowIds <- store.add(processedSources)
+    } yield workflowIds
+  }
+
+  private def processSources(sources: NonEmptyList[WorkflowSourceFiles],
+                             processOptions: WorkflowOptions => WorkflowOptionsJson):
+  Try[NonEmptyList[WorkflowSourceFiles]] = {
+    val nelTries: NonEmptyList[Try[WorkflowSourceFiles]] = sources map processSource(processOptions)
+    val seqTries: Seq[Try[WorkflowSourceFiles]] = nelTries.list.toList
+    val trySeqs: Try[Seq[WorkflowSourceFiles]] = TryUtil.sequence(seqTries)
+    import scalaz.Scalaz._
+    val tryNel: Try[NonEmptyList[WorkflowSourceFiles]] = trySeqs.map(_.toList.toNel.get)
+    tryNel
```

Comment:
ToL: Also open to folks helping with this HoF soup. Of course will be a different brand of soup for 0.22_hotfix and develop due to scalaz->cats.